File: clients/algoliasearch-client-java-2/algoliasearch-core/com/algolia/model/search/DictionaryEntry.java
package com.algolia.model.search;

import com.google.gson.annotations.SerializedName;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Objects;

/** A dictionary entry. */
public class DictionaryEntry extends HashMap<String, Object> {

  @SerializedName("objectID")
  private String objectID;

  @SerializedName("language")
  private String language;

  @SerializedName("word")
  private String word;

  @SerializedName("words")
  private List<String> words = null;

  @SerializedName("decomposition")
  private List<String> decomposition = null;

  @SerializedName("state")
  private DictionaryEntryState state = DictionaryEntryState.ENABLED;

  public DictionaryEntry setObjectID(String objectID) {
    this.objectID = objectID;
    return this;
  }

  /**
   * Unique identifier of the object.
   *
   * @return objectID
   */
  @javax.annotation.Nonnull
  public String getObjectID() {
    return objectID;
  }

  public DictionaryEntry setLanguage(String language) {
    this.language = language;
    return this;
  }

  /**
   * Language ISO code supported by the dictionary (e.g., \"en\" for English).
   *
   * @return language
   */
  @javax.annotation.Nonnull
  public String getLanguage() {
    return language;
  }

  public DictionaryEntry setWord(String word) {
    this.word = word;
    return this;
  }

  /**
   * The word of the dictionary entry.
   *
   * @return word
   */
  @javax.annotation.Nullable
  public String getWord() {
    return word;
  }

  public DictionaryEntry setWords(List<String> words) {
    this.words = words;
    return this;
  }

  public DictionaryEntry addWordsItem(String wordsItem) {
    if (this.words == null) {
      this.words = new ArrayList<>();
    }
    this.words.add(wordsItem);
    return this;
  }

  /**
   * The words of the dictionary entry.
   *
   * @return words
   */
  @javax.annotation.Nullable
  public List<String> getWords() {
    return words;
  }

  public DictionaryEntry setDecomposition(List<String> decomposition) {
    this.decomposition = decomposition;
    return this;
  }

  public DictionaryEntry addDecompositionItem(String decompositionItem) {
    if (this.decomposition == null) {
      this.decomposition = new ArrayList<>();
    }
    this.decomposition.add(decompositionItem);
    return this;
  }

  /**
   * A decomposition of the word of the dictionary entry.
   *
   * @return decomposition
   */
  @javax.annotation.Nullable
  public List<String> getDecomposition() {
    return decomposition;
  }

  public DictionaryEntry setState(DictionaryEntryState state) {
    this.state = state;
    return this;
  }

  /**
   * Get state
   *
   * @return state
   */
  @javax.annotation.Nullable
  public DictionaryEntryState getState() {
    return state;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    DictionaryEntry dictionaryEntry = (DictionaryEntry) o;
    return (
      Objects.equals(this.objectID, dictionaryEntry.objectID) &&
      Objects.equals(this.language, dictionaryEntry.language) &&
      Objects.equals(this.word, dictionaryEntry.word) &&
      Objects.equals(this.words, dictionaryEntry.words) &&
      Objects.equals(this.decomposition, dictionaryEntry.decomposition) &&
      Objects.equals(this.state, dictionaryEntry.state) &&
      super.equals(o)
    );
  }

  @Override
  public int hashCode() {
    return Objects.hash(
      objectID,
      language,
      word,
      words,
      decomposition,
      state,
      super.hashCode()
    );
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class DictionaryEntry {\n");
    sb.append("    ").append(toIndentedString(super.toString())).append("\n");
    sb.append("    objectID: ").append(toIndentedString(objectID)).append("\n");
    sb.append("    language: ").append(toIndentedString(language)).append("\n");
    sb.append("    word: ").append(toIndentedString(word)).append("\n");
    sb.append("    words: ").append(toIndentedString(words)).append("\n");
    sb
      .append("    decomposition: ")
      .append(toIndentedString(decomposition))
      .append("\n");
    sb.append("    state: ").append(toIndentedString(state)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: clients/algoliasearch-client-php/lib/Model/Search/DictionaryEntry.php
<?php

namespace Algolia\AlgoliaSearch\Model\Search;

/**
 * DictionaryEntry Class Doc Comment
 *
 * @category Class
 * @description A dictionary entry.
 *
 * @package Algolia\AlgoliaSearch
 */
class DictionaryEntry extends \Algolia\AlgoliaSearch\Model\AbstractModel implements ModelInterface, \ArrayAccess, \JsonSerializable
{
    /**
      * Array of property to type mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $modelTypes = [
        'objectID' => 'string',
        'language' => 'string',
        'word' => 'string',
        'words' => 'string[]',
        'decomposition' => 'string[]',
        'state' => '\Algolia\AlgoliaSearch\Model\Search\DictionaryEntryState',
    ];

    /**
      * Array of property to format mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $modelFormats = [
        'objectID' => null,
        'language' => null,
        'word' => null,
        'words' => null,
        'decomposition' => null,
        'state' => null,
    ];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelTypes()
    {
        return self::$modelTypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelFormats()
    {
        return self::$modelFormats;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'objectID' => 'setObjectID',
        'language' => 'setLanguage',
        'word' => 'setWord',
        'words' => 'setWords',
        'decomposition' => 'setDecomposition',
        'state' => 'setState',
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'objectID' => 'getObjectID',
        'language' => 'getLanguage',
        'word' => 'getWord',
        'words' => 'getWords',
        'decomposition' => 'getDecomposition',
        'state' => 'getState',
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     */
    public function __construct(array $data = null)
    {
        if (isset($data['objectID'])) {
            $this->container['objectID'] = $data['objectID'];
        }
        if (isset($data['language'])) {
            $this->container['language'] = $data['language'];
        }
        if (isset($data['word'])) {
            $this->container['word'] = $data['word'];
        }
        if (isset($data['words'])) {
            $this->container['words'] = $data['words'];
        }
        if (isset($data['decomposition'])) {
            $this->container['decomposition'] = $data['decomposition'];
        }
        if (isset($data['state'])) {
            $this->container['state'] = $data['state'];
        }
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        if (!isset($this->container['objectID']) || $this->container['objectID'] === null) {
            $invalidProperties[] = "'objectID' can't be null";
        }
        if (!isset($this->container['language']) || $this->container['language'] === null) {
            $invalidProperties[] = "'language' can't be null";
        }

        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }

    /**
     * Gets objectID
     *
     * @return string
     */
    public function getObjectID()
    {
        return $this->container['objectID'] ?? null;
    }

    /**
     * Sets objectID
     *
     * @param string $objectID unique identifier of the object
     *
     * @return self
     */
    public function setObjectID($objectID)
    {
        $this->container['objectID'] = $objectID;

        return $this;
    }

    /**
     * Gets language
     *
     * @return string
     */
    public function getLanguage()
    {
        return $this->container['language'] ?? null;
    }

    /**
     * Sets language
     *
     * @param string $language Language ISO code supported by the dictionary (e.g., \"en\" for English).
     *
     * @return self
     */
    public function setLanguage($language)
    {
        $this->container['language'] = $language;

        return $this;
    }

    /**
     * Gets word
     *
     * @return string|null
     */
    public function getWord()
    {
        return $this->container['word'] ?? null;
    }

    /**
     * Sets word
     *
     * @param string|null $word the word of the dictionary entry
     *
     * @return self
     */
    public function setWord($word)
    {
        $this->container['word'] = $word;

        return $this;
    }

    /**
     * Gets words
     *
     * @return string[]|null
     */
    public function getWords()
    {
        return $this->container['words'] ?? null;
    }

    /**
     * Sets words
     *
     * @param string[]|null $words the words of the dictionary entry
     *
     * @return self
     */
    public function setWords($words)
    {
        $this->container['words'] = $words;

        return $this;
    }

    /**
     * Gets decomposition
     *
     * @return string[]|null
     */
    public function getDecomposition()
    {
        return $this->container['decomposition'] ?? null;
    }

    /**
     * Sets decomposition
     *
     * @param string[]|null $decomposition a decomposition of the word of the dictionary entry
     *
     * @return self
     */
    public function setDecomposition($decomposition)
    {
        $this->container['decomposition'] = $decomposition;

        return $this;
    }

    /**
     * Gets state
     *
     * @return \Algolia\AlgoliaSearch\Model\Search\DictionaryEntryState|null
     */
    public function getState()
    {
        return $this->container['state'] ?? null;
    }

    /**
     * Sets state
     *
     * @param \Algolia\AlgoliaSearch\Model\Search\DictionaryEntryState|null $state state
     *
     * @return self
     */
    public function setState($state)
    {
        $this->container['state'] = $state;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param int $offset Offset
     *
     * @return bool
     */
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param int $offset Offset
     *
     * @return mixed|null
     */
    public function offsetGet($offset)
    {
        return $this->container[$offset] ?? null;
    }

    /**
     * Sets value based on offset.
     *
     * @param int|null $offset Offset
     * @param mixed    $value  Value to be set
     *
     * @return void
     */
    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param int $offset Offset
     *
     * @return void
     */
    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }
}

